### PR TITLE
gobin: take into account package replacements

### PR DESCRIPTION
**Pull request by hdonnay**

### CODE REVIEW

File: gobin/exe.go [hunk 1]
```diff
@@ -132,6 +132,9 @@ func toPackages(ctx context.Context, out *[]*claircore.Package, p string, r io.R
 		vs[bi.Main.Path] = bi.Main.Version
 	}
 	for _, d := range bi.Deps {
+		if r := d.Replace; r != nil {
```
Comment (crozzy) [inline gobin/exe.go:135]: Is it possible the tree is bigger than 2 levels deep? Like can you replace the right-hand side of a replace with another replace? (I know this a little like a play stupid games win stupid prizes issue)

```
replace (
    golang.org/x/net v1.2.3 => example.com/fork/net v1.4.5
    example.com/fork/net v1.4.5 => example.com/another_fork v1.4.6
)
```

**Review by crozzy — CHANGES_REQUESTED** [review]
One query, also, you think it merits a version increment?

File: gobin/exe.go (hunk 1 above)
Comment (hdonnay) [reply gobin/exe.go:135]: I'll look into that. I assume the whole chain is resolved, but don't know that

**Comment by dominikholler** [comment]
@hdonnay thanks for the PR!

**Review by crozzy — APPROVED** [review]
LGTM, thanks for checking

**Pull request merged**